flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MysqlPipelineNewlyAddedTableTCase.java

```diff
@@ -0,0 +1,478 @@
+import static org.apache.flink.cdc.connectors.mysql.testutils.MySqSourceTestUtils.TEST_PASSWORD;
+import static org.apache.flink.cdc.connectors.mysql.testutils.MySqSourceTestUtils.TEST_USER;
+import static org.apache.flink.cdc.connectors.mysql.testutils.MySqSourceTestUtils.fetchResults;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** IT tests to cover various newly added tables during capture process in pipeline mode. */
+public class MysqlPipelineNewlyAddedTableTCase extends MySqlSourceTestBase {
+    private final UniqueDatabase customDatabase =
+            new UniqueDatabase(MYSQL_CONTAINER, "customer", "mysqluser", "mysqlpw");
+
+    private final ScheduledExecutorService mockBinlogExecutor = Executors.newScheduledThreadPool(1);
+
+    @Before
+    public void before() throws SQLException {
+        TestValuesTableFactory.clearAllData();
+        customDatabase.createAndInitialize();
+
+        try (MySqlConnection connection = getConnection()) {
+            connection.setAutoCommit(false);
+            // prepare initial data for given table
+            String tableId = customDatabase.getDatabaseName() + ".produce_binlog_table";
+            connection.execute(
+                    format("CREATE TABLE %s ( id BIGINT PRIMARY KEY, cnt BIGINT);", tableId));
+            connection.execute(
+                    format("INSERT INTO  %s VALUES (0, 100), (1, 101), (2, 102);", tableId));
+            connection.commit();
+
+            // mock continuous binlog during the newly added table capturing process
+            mockBinlogExecutor.schedule(
+                    () -> {
+                        try {
+                            connection.execute(
+                                    format("UPDATE  %s SET  cnt = cnt +1 WHERE id < 2;", tableId));
+                            connection.commit();
+                        } catch (SQLException e) {
+                            e.printStackTrace();
+                        }
+                    },
+                    500,
+                    TimeUnit.MICROSECONDS);
+        }
+    }
+
+    @After
+    public void after() {
+        mockBinlogExecutor.shutdown();
+    }
+
+    private MySqlConnection getConnection() {
+        Map<String, String> properties = new HashMap<>();
+        properties.put("database.hostname", MYSQL_CONTAINER.getHost());
+        properties.put("database.port", String.valueOf(MYSQL_CONTAINER.getDatabasePort()));
+        properties.put("database.user", customDatabase.getUsername());
+        properties.put("database.password", customDatabase.getPassword());
+        properties.put("database.serverTimezone", ZoneId.of("UTC").toString());
+        io.debezium.config.Configuration configuration =
+                io.debezium.config.Configuration.from(properties);
+        return DebeziumUtils.createMySqlConnection(configuration, new Properties());
+    }
+
+    @Test
+    public void testAddNewTableOneByOne() throws Exception {
```

Comment:
Great catch! 
Could you retract a common method like what NewlyAddedTableITCase#testNewlyAddedTableOneByOne does, so we can test different parallelism, different numbers of tables and different failover later?
And it will be better to add testRemoveTablesOneByOne.

Reply:
Well, I'll do it later.